pkg/kubectl-argo-rollouts/cmd/create/create.go

```diff
@@ -0,0 +1,279 @@
+	"io/ioutil"
+	"strings"
+	"unicode"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts"
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/cmd/get"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+	analysisutil "github.com/argoproj/argo-rollouts/utils/analysis"
+)
+
+type CreateOptions struct {
+	get.GetOptions
+	options.ArgoRolloutsOptions
+
+	Files    []string
+	From     string
+	FromFile string
+}
+
+type CreateAnalysisRunOptions struct {
+	options.ArgoRolloutsOptions
+
+	Name         string
+	GenerateName string
+	ArgFlags     []string
+	From         string
+	FromFile     string
+}
+
+// NewCmdCreate returns a new instance of an `rollouts create` command
+func NewCmdCreate(o *options.ArgoRolloutsOptions) *cobra.Command {
+	createOptions := CreateOptions{
+		ArgoRolloutsOptions: *o,
+	}
+	var cmd = &cobra.Command{
+		Use:   "create -f my-experiment.yaml",
+		Short: "Create a rollout resource",
+		Example: o.Example(`
+  # Create a resource and watch it
+  %[1]s create -f my-experiment.yaml -w
+`),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(createOptions.Files) == 0 {
+				return o.UsageErr(c)
+			}
+			if len(createOptions.Files) > 1 && createOptions.Watch {
+				return errors.New("Cannot watch multiple resources")
+			}
+
+			var objs []runtime.Object
+			for _, f := range createOptions.Files {
+				obj, err := createOptions.createResource(f)
+				if err != nil {
+					return err
+				}
+				objs = append(objs, obj)
+			}
+			if createOptions.Watch {
+				switch obj := objs[0].(type) {
+				case *v1alpha1.Rollout:
+					getCmd := get.NewCmdGetRollout(o)
+					getCmd.SetArgs([]string{obj.Name, "--watch"})
+					return getCmd.Execute()
+				case *v1alpha1.Experiment:
+					getCmd := get.NewCmdGetExperiment(o)
+					getCmd.SetArgs([]string{obj.Name, "--watch"})
+					return getCmd.Execute()
+				default:
+					return errors.New("Can only watch resources of type Rollout or Experiment")
+				}
+			}
+			return nil
+		},
+	}
+	cmd.AddCommand(NewCmdCreateAnalysisRun(o))
+
+	o.AddKubectlFlags(cmd)
+	cmd.Flags().StringArrayVarP(&createOptions.Files, "filename", "f", []string{}, "Files to use to create the resource")
+	cmd.Flags().BoolVarP(&createOptions.Watch, "watch", "w", false, "Watch live updates to the resource after creating")
+	cmd.Flags().BoolVar(&createOptions.NoColor, "no-color", false, "Do not colorize output")
+	return cmd
+}
+
+// isJSON detects if the byte array looks like json, based on the first non-whitespace character
+func isJSON(fileBytes []byte) bool {
+	for _, b := range fileBytes {
+		if !unicode.IsSpace(rune(b)) {
+			return b == '{'
+		}
+	}
+	return false
+}
```

Comment:
I'm surprised there isn't an easier way of doing this with the libraries we imported.

Reply:
That being said, I only do a quick search to see if I find one, but otherwise, I think it's fine